Shared/Samples/Add custom dynamic entity data source/AddCustomDynamicEntityDataSourceView.swift

```diff
@@ -0,0 +1,164 @@
+                        ForEach(attributes.sorted(by: { $0.key < $1.key }), id: \.key) { item in
+                            Text("\(item.key): \(String(describing: item.value))")
+                        }
+                    }
+                }
+                .task(id: tappedScreenPoint) {
+                    guard let tappedScreenPoint,
+                          let identifyResult = try? await proxy.identify(
+                            on: dynamicEntityLayer,
+                            screenPoint: tappedScreenPoint,
+                            tolerance: 2
+                          ) else {
+                        // Hides the callout.
+                        calloutPlacement = nil
+                        
+                        return
+                    }
+                    
+                    // Set the callout placement to the observation that was tapped on.
+                    calloutPlacement = identifyResult.geoElements.first.map { .geoElement($0) }
+                }
+        }
+    }
+}
+
+/// The vessel feed that is emitting custom dynamic entity events.
+private struct VesselFeed: CustomDynamicEntityFeed {
+    let events: AsyncThrowingStream<CustomDynamicEntityFeedEvent, Error> = .init { continuation in
+        Task.detached {
+            do {
+                let fileHandle = try FileHandle(forReadingFrom: .selectedVesselsDataSource)
+                let decoder = JSONDecoder()
+                
+                // Loop through each line in the JSON file.
+                for try await line in fileHandle.bytes.lines {
+                    // Delay observations to simulate live data.
+                    try await Task.sleep(nanoseconds: 10_000_000)
+                    
+                    let decodable = try decoder.decode(
+                        AddCustomDynamicEntityDataSourceView.Vessel.self,
+                        from: line.data(using: .utf8)!
+                    )
+                    
+                    // The geometry that was decoded from the JSON.
+                    let geometry = decodable.geometry
+                    
+                    // We successfully decoded the vessel JSON so we should
+                    // add that vessel as a new observation.
+                    continuation.yield(.newObservation(
+                        geometry: Point(x: geometry.x, y: geometry.y, spatialReference: .wgs84),
+                        attributes: decodable.attributes
+                    ))
+                }
+                
+                continuation.finish()
+            } catch {
+                continuation.finish(throwing: error)
+            }
+        }
+    }
+}
+
+private extension Array<Field> {
+    /// An array of fields that match the attributes of each observation in the data source.
+    ///
+    /// This schema is derived from the first row in the custom data source.
+    static var vesselFields: Self {
```

Comment:
Maybe move it to where the data source info is created, to make it more obvious as one of the steps to create the data source.